lightkurve/targetpixelfile.py

```diff
@@ -190,6 +193,85 @@ def from_archive(target, cadence='long', quarter=None, month=None,
     def __repr__(self):
         return('KeplerTargetPixelFile Object (ID: {})'.format(self.keplerid))
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: 'KIC', 'EPIC' or 'Gaia'
+            Catalog to query. For Kepler target pixel files the default is 'KIC',
+            for K2 target pixel files the default is 'EPIC'.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18 for both Kepmag and Gmag.
+        dist_tolerance: float
+            Maximum distance (in arcseconds) a source may be separated from the edge
+            of the target pixel file.
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+            ID : Catalog ID from catalog.
+            RAJ200: Right ascension [degrees]
+            DEJ2000: Declination [degrees]
+            pmRA: Proper motion for right ascension [mas/year]
+            pmDEC: Proper motion for declination [mas/year]
+            Kpmag: Magnitude in Kepler band [mag]
+
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                raise ValueError('Please provide a catalog.')
+
+        if dist_tolerance < 3:
+            log.warning('Distance tolerance is too low')
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        #Find the size of the TPF
+        radius = ( (np.max(self.flux.shape[1:2]) * 4) + dist_tolerance)/ 60.0
+
+        # query around centre with radius
+        data = query_catalog(cent, radius=radius, catalog=catalog)
+
+        # Find where nans are in cadence
+        find_nans = np.isfinite(self.flux)
+        sum_nans = np.sum(find_nans, axis=0)
+
+        # Load ra & dec of all tpf pixels
+        pixels_ra, pixels_dec = self.get_coordinates(cadence=int(len(self.cadenceno)/2))
+
+        # Load pixel ra, dec with no nans
+        pixel_radec = np.asarray([pixels_ra[sum_nans != 0].ravel(), pixels_dec[sum_nans != 0].ravel()])
+
+        # Make pixel pairs into SkyCoord
+        sky_pixel_pairs = SkyCoord(ra=pixel_radec[0], dec=pixel_radec[1], frame='icrs', unit=(u.deg, u.deg))
+        # Make pairs in sky_sources
+        sky_sources = SkyCoord(ra=data['ra'], dec=data['dec'], frame='icrs', unit=(u.deg, u.deg))
+
+        separation_mask = []
```

Comment:
Rather than have a list that you make into an array in line 268 you could use

```
separation_mask = np.zeros(len(data), dtype=bool)
for idx in range (len(data)):
            s = sky_pixel_pairs.separation(sky_sources[i])  # Estimate separation
            separation = np.any(s.arcsec <= dist_tolerance)
            separation_mask[idx] = separation
```

Note I've renamed `i` to `idx` because it's slightly easier to understand what `idx` is, and it's easier to read later when we have to debug.